File: GeeksForGeeks/reverseWords.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends

class Solution
{
    public:
    //Function to reverse words in a given string.
    string reverseWords(string S) 
    { 
        stack<string> pil;
        for (size_t i = 0; i < S.size(); i++)
        {
            string tmp = "";
            while (i < S.size() && S[i] != '.')
            {
                tmp += S[i++];
            }
            pil.push(tmp);
        }
        if (S[S.size()-1] == '.')
            pil.push("");
        string result = pil.top();
        pil.pop();
        while (pil.size() > 0)
        {
            result += "." + pil.top();
            pil.pop();
        }
        return result;
    } 
};

//{ Driver Code Starts.
int main() 
{
    int t;
    cin >> t;
    while (t--) 
    {
        string s;
        cin >> s;
        Solution obj;
        cout<<obj.reverseWords(s)<<endl;
    }
}
// } Driver Code Ends

File: LeetCode/reverseWords.cs
// 1/10/2023

public class Solution {
    public string ReverseWords(string s) {
        var words = s.Split(' ');
        string result = "";
        for (int j = 0; j < words.Length; j++)
        {
            string reverse = "";
            for (int i = 0; i < words[j].Length; i++)
            {
                reverse = words[j][i] + reverse;
            }
            result += reverse;
            if (2*j + 1 < 2*words.Length-1)
                result += " ";
        }
        return result;
    }
}